src/Compilers/Core/Portable/Compilation/SemanticModel.cs

```diff
@@ -78,7 +78,7 @@ public SyntaxTree SyntaxTree
             {
                 return GetOperationCore(node, cancellationToken);
             }
-            catch (Exception e) when (FatalError.ReportAndCatchUnlessCanceled(e))
+            catch (Exception e) when (FatalError.ReportAndCatchUnlessCanceled(e, cancellationToken))
```

Comment:
>cancellationToken [](start = 81, length = 17)

What would be the purpose of cancelling while reporting a crash?

Reply:
> What would be the purpose of cancelling while reporting a crash?

Throwing an exception during the execution of an exception filter will FailFast, so this could never have semantics like that. This new overload verifies that methods only report cancellation in cases where cancellation was expected by the caller.

Reply:
I expect if we do that, it will be like `[MaybeNullWhen]`. We started using a named parameter, but quickly realized it's _always_ the same name parameter for that call and stopped.

Reply:
> We started using a named parameter, but quickly realized it's always the same name parameter for that call and stopped.

I am not following. I am suggesting to change the name of the parameter for the ```ReportAndCatchUnlessCanceled``` API to make it clear that its execution is not going to throw "cancelled" exception because of the token.


---
In reply to: [604180763](https://github.com/dotnet/roslyn/pull/52247#discussion_r604180763) [](ancestors = 604180763)

Reply:
I'm not sure what other name to use. We shouldn't use `expectedCancellationToken`, because it suggests that the method will check that the [`OperationCanceledException.CancellationToken`](https://docs.microsoft.com/en-us/dotnet/api/system.operationcanceledexception.cancellationtoken?view=net-5.0#System_OperationCanceledException_CancellationToken) property matches the expected token, but this also not what the method does.

Reply:
> I'm not sure what other name to use. We shouldn't use expectedCancellationToken, because it suggests that the method will check that the [OperationCanceledException.CancellationToken](https://docs.microsoft.com/en-us/dotnet/api/system.operationcanceledexception.cancellationtoken?view=net-5.0#System_OperationCanceledException_CancellationToken) property matches the expected token, but this also not what the method does.

Perhaps we should start with clearly describing what the method does. The current doc comment on it doesn't do a good job in that, in my opinion. After that, we can come up with names and quite possible rename the method itself if that would help.


---
In reply to: [604189805](https://github.com/dotnet/roslyn/pull/52247#discussion_r604189805) [](ancestors = 604189805)

Reply:
@AlekseyTs see #52253

Reply:
perhaps a name change to `ReportAndCatchUnlessCurrentOperationCanceled` to make the distinction clear?